test/src/kbasesearchengine/test/main/NarrativeInfoDecoratorTest.java

```diff
@@ -372,6 +377,66 @@ public void listTypes() throws Exception {
                 "type", dummyOutput)));
     }
 
+    @Test
+    public void searchObjectsWithDeletedWs() throws Exception {
+        final NarrativeInfoDecorator nid = setUpSearchObjectsWithDeletedWs();
+        final Pagination pag = new Pagination().withCount(3L).withStart(0L);
+        final SearchObjectsInput dummyInput = new SearchObjectsInput()
+                .withPostProcessing(new PostProcessing().withAddNarrativeInfo(1L))
+                .withPagination(pag);
+
+        final ArrayList<ObjectData> expectedObjs = new ArrayList<>();
+        expectedObjs.add(new ObjectData().withGuid("WS:4/1/7").withCreator("user"));
+        expectedObjs.add(new ObjectData().withGuid("WS:4/2/1").withCreator("user"));
+        expectedObjs.add(new ObjectData().withGuid("WS:4/1/7").withCreator("user"));
+
+        final SearchObjectsOutput res = nid.searchObjects(dummyInput, "user");
+
+        compareSearchObjectOutputRes(res.getObjects(), expectedObjs);
+        assertThat("incorrect number of hits looked at", res.getTotal(), is(6L));
```

Comment:
Total should be 3 since there are only three objects being returned in res?

Reply:
changing total to be the total number of objects looked at so the ones removed won't mess up pagination